nni/experiment/config/utils/internal.py

```diff
@@ -198,3 +203,33 @@ def get_ipv4_address() -> str:
     addr = s.getsockname()[0]
     s.close()
     return addr
+
+def load_experiment_config(config_json: dict) -> ExperimentConfig | RetiariiExeConfig:
+    # avoid circular import
+    from nni.nas.experiment.config import RetiariiExeConfig
+    from nni.nas.experiment.pytorch import RetiariiExperiment
+    from ..experiment_config import ExperimentConfig
+    from ...experiment import Experiment
+    exp_cls = get_experiment_cls_using_config(config_json)
+    if exp_cls is Experiment:
```

Comment:
If you raise exception in `get_experiment_cls_using_config`, this part can be `get_experiment_cls_using_config(config_json)(**config_json)`

Reply:
good point! but the required is config's cls not experiment's cls. updated the code following your suggestion.